tests/test_version.py

```diff
@@ -1,28 +1,20 @@
 import os
 import unittest
+from unittest.mock import patch
 
 import pystac
 from tests.utils import TestCases
 
 
 class VersionTest(unittest.TestCase):
     def setUp(self) -> None:
-        self._prev_env_version = os.environ.get("PYSTAC_STAC_VERSION_OVERRIDE")
-        self._prev_version = pystac.get_stac_version()
-
-    def tearDown(self) -> None:
-        if self._prev_env_version is None:
-            os.environ.pop("PYSTAC_STAC_VERSION_OVERRIDE", None)
-        else:
-            os.environ["PYSTAC_STAC_VERSION_OVERRIDE"] = self._prev_env_version
-        pystac.set_stac_version(None)
+        pystac.version.STACVersion._override_version = None
 
     def test_override_stac_version_with_environ(self) -> None:
-
         override_version = "1.0.0-gamma.2"
-        os.environ["PYSTAC_STAC_VERSION_OVERRIDE"] = override_version
-        cat = TestCases.test_case_1()
-        d = cat.to_dict()
+        with patch.dict(os.environ, {"PYSTAC_STAC_VERSION_OVERRIDE": override_version}):
```

Comment:
I've generally preferred using this as a function decorator instead of a context manager in the past because it ensures that the variable gets set for the entire duration of that test and simplifies the function code.

The context manager works just fine in this case, but I just wanted to point that out as an alternative in case we want to establish it as a pattern.

Reply:
That's a good way of thinking about it. I'll just explain my reasoning since it's quite possible we have some common ground here. Basically, I try to minimize the size of basically any scope, including `with`, `@decorators`, `try…except` and the like. Each scope is another thing to keep in mind when reading each statement, so I feel like this minimizes the amount of context the reader has to keep in mind when reading.

Reply:
Yeah, that all makes good sense to me! Let's keep this as-is.